### DISCUSSION TITLE
Expose `columnProps` in RAC `Cell`

### DISCUSSION THREAD

**Discussion by sebald** [adobe/react-spectrum#5481] [Ideas]
Hey,

I know that the RAC component should cover most cases and we should reach for the hooks if we want more customization, but I found that a lot of common features (e.g. column alignment, striped columns, ...) are enable by exposing the properties of the columns to cell in the same column.

Internally the `TableCell` already has access to the collection. Exposing the respective column of a cell should be straight forward (happy to submit a PR!)

https://github.com/adobe/react-spectrum/blob/48f29e1dc686f84b13a08c991d257b1719a12925/packages/react-aria-components/src/Table.tsx#L1135

Passing something like the `state.collection.columns[cell.index].props` as `columnProps` to a render function (via `values`) would greatly improve the versatility of the `Table` component.

(Context: I tried to use the `TableStateContext`, but the cell doesn't know its own index.)

**Comment by sebald** [comment]
Not sure if this is a good idea anymore. The `<Table>` actually can get the state when using a function with `className` or `style`. So maybe, for consistency reasons, this is also possible to do for the other table related components.

**Comment by sergiocarneiro** [comment]
Any updates on this? I'd also like to add styling props to `Column` that can be shared with `Cell` — like React Spectrum does with `align` and `showDivider`.
